service/src/main/java/com/codeforcommunity/processor/ImportProcessorImpl.java

```diff
@@ -129,4 +137,115 @@ public void importReservations(
       record.store();
     }
   }
+
+  private void importSiteEntryUsername(Integer siteEntryId, String username) {
+    EntryUsernamesRecord record = db.newRecord(ENTRY_USERNAMES);
+
+    record.setEntryId(siteEntryId);
+    record.setUsername(username);
+
+    record.store();
+  }
+
+  @Override
+  public void importSites(JWTData userData, ImportSitesRequest importSitesRequest) {
+    if (userData.getPrivilegeLevel() != PrivilegeLevel.SUPER_ADMIN) {
+      throw new AuthException("User does not have the required privilege level.");
+    }
+    if ((!db.fetchExists(db.selectFrom(USERS).where(USERS.ID.eq(userData.getUserId()))))) {
+      throw new UserDoesNotExistException(userData.getUserId());
+    }
+
+    List<SitesRecord> sitesRecords = new ArrayList<>();
+    List<Map.Entry<SiteEntriesRecord, String>> pairList = new ArrayList<>();
```

Comment:
Would be good practice to call this something more informative `siteEntryRecordsAndUsername` for example